### PR TITLE
Add version to output

### LINKED ISSUE #425: add version to all relevant models + seeds
### Describe the feature

We should add the model version to all outputs to disambiguate model names in the case where multiple versions of a model do not conform to a rule. 

This is a prerequisite to #424 

### Are you interested in contributing this feature?
Yes I am (already started)

### PR DESCRIPTION
**Pull request by dave-connors-3**
Closes #425 

This is a:
- [ ] bug fix PR with no breaking changes
- [x] new functionality

## Link to Issue 

This PR 

- adds `version` and `deprecation_date` to our unpack macros
- updates the resource names to include the version if present -- for example `model.my_project.my_model.v1` would have been identified as `my_model` in all rules in the project before this change, and will be identified as `my_model.v1` after this change. `model.my_project.my_unversioned_model` would be identified as `my_unversioned_model` before and after this change

## Description & motivation

This is a more correct representation of the nodes in the project, and disambiguates situations where more than one version of a model violates a rule, or when a versioned model is referenced only by name in a rule output

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
    - [ ] Databricks
    - [x] DuckDB
    - [ ] Trino/Starburst
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

**Comment by b-per** [comment]
DuckDB 0.10 is out as well but the bug on the check still exists on the new version

**Comment by dave-connors-3** [comment]
@b-per @graciegoheen looks like the only failure here is another weird decimal rounding issue in the test:

```
    {
      "which_diff": "a_minus_b",
      "total_models": 18,
      "total_tests": 13,
      "tested_models": 6,
      "test_coverage_pct": 33.33000183105469,
      "test_to_model_ratio": 0.7222,
      "staging_test_coverage_pct": 80.0,
      "intermediate_test_coverage_pct": 25.0,
      "marts_test_coverage_pct": 0.0,
      "other_test_coverage_pct": 25.0
    },
    {
      "which_diff": "b_minus_a",
      "total_models": 18,
      "total_tests": 13,
      "tested_models": 6,
      "test_coverage_pct": 33.33,
      "test_to_model_ratio": 0.7222,
      "staging_test_coverage_pct": 80.0,
      "intermediate_test_coverage_pct": 25.0,
      "marts_test_coverage_pct": 0.0,
      "other_test_coverage_pct": 25.0
    }
  ]
}
```

not sure it's worth fretting about or if this one can just get merged in!

**Comment by b-per** [comment]
@dave, could you treat databricks and trino the same way we did for DuckDB so that this particular column is not tested?
That way, CI should continue passing.

**Comment by dave-connors-3** [comment]
@b-per done!

File: models/marts/core/int_all_graph_resources.sql [hunk 1]
```diff
@@ -34,6 +34,7 @@ unioned_with_calc as (
         *,
         case 
             when resource_type = 'source' then  {{ dbt.concat(['source_name',"'.'",'name']) }}
+            when version is not null then {{ dbt.concat(['name',"'.v'",'version']) }}
```
Comment (graciegoheen) [inline models/marts/core/int_all_graph_resources.sql:37]: Why this vs. a new field called "version"?

Comment (dave-connors-3) [reply models/marts/core/int_all_graph_resources.sql:37]: i add that on line 87!

Comment (graciegoheen) [reply models/marts/core/int_all_graph_resources.sql:37]: ope - yes, thank you!

**Review by graciegoheen — APPROVED** [review]

**Pull request merged**